packages/core/src/core/plan/group.ts

```diff
@@ -51,17 +51,34 @@ export const getCustomGroupIds = async (
   if (Object.keys(customGroupIdFunctions).length === 0) {
     return { changeGroupIdMap: new Map() }
   }
-  const changesPerAdapter = collections.iterable.groupBy(
+  const changesPerAccount = collections.iterable.groupBy(
     wu(changes.keys()).map(id => ({ id, change: changes.getData(id) })),
     ({ change }) => getChangeData(change).elemID.adapter,
   )
-  const changeGroupInfoPerAdapter = await awu(changesPerAdapter.entries())
-    .filter(([adapterName]) => adapterName in customGroupIdFunctions)
-    .map(([name, adapterChanges]) => (
-      customGroupIdFunctions[name](new Map(adapterChanges.map(({ id, change }) => [id, change])))
+  const changeGroupInfoPerAccount = await awu(changesPerAccount.entries())
+    .filter(([accountName]) => accountName in customGroupIdFunctions)
+    .map(([accountName, accountChanges]) => (
+      customGroupIdFunctions[accountName](new Map(accountChanges.map(({ id, change }) => [id, change])))
+        .then(({ changeGroupIdMap, disjointGroups }) => {
+          log.debug(`Adding account name prefix to change group ids for account ${accountName}`)
+          const withAccountNamePrefix = (groupId: ChangeGroupId): ChangeGroupId => `${accountName}__${groupId}`
+          const changeGroupIdMapWithAccountNamePrefix = new Map<ChangeId, ChangeGroupId>()
+          changeGroupIdMap.forEach((groupId, changeId) => {
+            changeGroupIdMapWithAccountNamePrefix.set(
+              changeId,
+              withAccountNamePrefix(groupId)
+            )
+          })
+          return {
+            changeGroupIdMap: changeGroupIdMapWithAccountNamePrefix,
+            disjointGroups: disjointGroups
+              ? new Set(Array.from(disjointGroups).map(withAccountNamePrefix))
+              : undefined,
+          }
+        })
```

Comment:
we should not use `.then` when we can use `await`, also, unless I am missing something this can be done with significantly less code

```suggestion
    .map(async ([accountName, accountChanges]) => (
      const withAccountNamePrefix = (groupId: ChangeGroupId): ChangeGroupId => `${accountName}/${groupId}`
      const { changeGroupIdMap, disjointGroups } = await customGroupIdFunctions[name](
        new Map(adapterChanges.map(({ id, change }) => [id, change]))
      )
      return {
        changeGroupIdMap: new Map(wu(changeGroupIdMap.entries()).map(([changeId, groupId]) => [changeId, withAccountNamePrefix(groupId)])),
        disjointGroups: disjointGroups === undefined ? undefined : new Set(wu(disjointGroups).map(withAccountNamePrefix))
      }
```